pkg/build/api/v1/types.go

```diff
@@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
 
 	// Git is the git information if the Type is BuildSourceGit
 	Git *GitInfo `json:"git,omitempty"`
+
+	// Env contains additional environment variables you want to pass into a builder container
+	Env []kapi.EnvVar `json:"env,omitempty"`
```

Comment:
On Tue, Apr 26, 2016 at 12:37 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/api/v1/types.go
> <https://github.com/openshift/origin/pull/8477#discussion_r61119692>:
>
> > @@ -620,6 +620,9 @@ type GenericWebHookEvent struct {
> >
> >  	// Git is the git information if the Type is BuildSourceGit
> >  	Git *GitInfo `json:"git,omitempty"`
> > +
> > +	// Env contains additional environment variables you want to pass into a builder container
> > +	Env []kapi.EnvVar `json:"env,omitempty"`
>
> Do we call this "Env" in any other place?
>
> yes:
> https://github.com/openshift/origin/blob/master/pkg/build/api/v1/types.go#L392-L393
>
> I would sort of expect the webhook trigger config to be able to limit what
> is accepted here
>
> interesting idea. no there is currently no way to constrain that. I could
> see there being a case for a AllowEnv field on the trigger definition in
> the buildconfig.
>
@bparees - just to be sure, are you referring to AllowEnv being a boolean
toggle we would add to
https://github.com/openshift/origin/blob/master/pkg/build/api/types.go#L573-L576
?

@liggit - were you referring to something other than an on/off switch?
some control on the content of the k/v pairs (though the only thing that
comes to my mind in that space is
perhaps allowing regular expressions to be defined which the keys/values
would then be matched against) ?

> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8477/files/fde0d585e7272ad1f545600043154b5afd9a9cd0#r61119692>
>